tests/lib/rules/jest/no-all-mocks-methods.js

```diff
@@ -0,0 +1,41 @@
+const {RuleTester} = require('eslint');
+const rule = require('../../../../lib/rules/jest/no-all-mocks-methods');
+
+const ruleTester = new RuleTester();
+function errorWithMethodName(name) {
+  return [
+    {
+      type: 'Identifier',
+      message: `Do not use ${name} or related methods that are not explicit to your test. Instead, target individual mocks.`,
+    },
+  ];
+}
+
+ruleTester.run('no-all-mocks-methods', rule, {
+  valid: [
+    {
+      code: `jest.mock()`,
+    },
+    {
+      code: `jest.fn()`,
```

Comment:
This rule is fine as is, but you could check that the method is chained off the `jest` object. Right now, if there were a method in your application called `clearAllMocks` (completely unrelated to jest) that would be considered invalid. This is highly unlikely, but you could alter this rule to prevent against it.

At the same time, if you solve that problem, a second problem could occur if the methods were destructured off the jest object (`const {clearAllMocks} = jest;` or `const clearAllMocks = jest.clearAllMocks` or even `somethingTotallyDifferent = jest.clearAllMocks`). I think your rule will catch this right now.

You could protect against all of the above at the same time, but it is a lot more complex and I don't anticipate too many problems with the rule as is. However, it would be a fun challenge. I am also more than happy to pair again on it.